### DISCUSSION TITLE
Deployment to Vercel does not work as my local server.

### DISCUSSION THREAD

**Discussion by benjimosso** [vercel/community#7363] [Help]
### Summary

Hi Everyone and thanks so much for your help! 
I have a (simple?) problem when I deploy my application to Vercel. 
context: 
the application is an auction app, I'm building it for a friend who is battling cancer and she is making an event and I though this would be cool for people to bid on things that local artist and businesses donate so we can raise money. 
Problem: 
the specific problem I have is when I check the people who won the auction... every item has a ending time, once the auction ends I check for the winners and send an email using resend. 
on my DB, items have a Boolean (sentEmail). Once the email is sent with Resend I fire a function to update that record from false to true. 
it works fine on my local server but when I deploy to Vercel the function that updates the Boolean does not seem to work.
any idea why is this happening? 
I'll be offline for a couple days (going to the mountains yay!) 

### Steps to Reproduce

Here is how I check who won:

```typescript
let Winners: {
  name: string;
  amount: number;
  itemName: string;
  email: string;
  itemId: number;
  itemImage: string;
}[] = [];

items.map((item: Item) => {
  if (item.endDate < new Date().toISOString()) {
    bids?.map((bid: Bids) => {
      if (
        item.id === bid.item_id &&
        item.currentBid === bid.amount &&
        !item.emailSent
      ) {
        Winners.push({
          name: bid.full_name,
          amount: bid.amount,
          itemId: item.id,
          email: bid.email,
          itemName: item.name,
          itemImage: item.image,
        });
      }
    });
  }
});
```

If there's winners, send each one an email:

```typescript
if (Winners.length > 0) {
  console.log("Winners", Winners, Winners.length);
  Winners.map(async (winner) => {
    sendEmail(
      winner.name,
      winner.itemName,
      winner.email,
      winner.itemId,
      winner.itemImage,
      winner.amount
    );
  });
}
```

Function that fires Resend and sends emails:

```typescript
async function sendEmail(
  name: string,
  itemName: string,
  email: string,
  itemId: number,
  itemImage: string,
  amount: number
) {
  const resend = new Resend(process.env.RESEND_API_KEY);

  const { data, error } = await resend.emails.send({
    from: "(here is where the domain goes.)",
    to: [email],
    subject: "You Won The Bid!",
    react: EmailTemplate({
      name,
      itemName,
      itemId,
      itemImage,
      amount,
    }) as React.ReactElement,
  });
  if (error) {
    return Response.json({ error }, { status: 500 });
  }
  if (data) {
    console.log("Email Sent", data);
    await emailSent({ itemId });
  }
}
```

Function to update the item status on the database to check if email was sent.

```typescript
export async function emailSent({ itemId }: { itemId: number }) {
  const supabase = createClient();
  const { data, error } = await supabase.from("items").update({ emailSent: true }).eq("id", itemId).select();
  if (error) {
    console.error(error);
  }
  console.log("Item ID", itemId);
  console.log("EmailSent Function Response:", data);
}
```

**Comment by pawlean** [comment]
Hi @benjimosso!

Could it be a similar issue as to [this](https://github.com/orgs/vercel/discussions/7347#discussioncomment-9935627)?

**Reply by benjimosso** [reply]
Hi @pawlean  Thanks so much for your response, sorry I'm late to reply I was in the mountains and didn't have internet access. 
I will check the link you sent me and let you know! thanks again!

**Reply by benjimosso** [reply]
@pawlean soo I figure it out, it's my fault and a mistake on the DB... sorry for wasting your time. I will marked this as closed.

**Comment by benjimosso** [comment]
RLS policy issue on my DataBase...

[a comment by an automated account is not shown]
